Crash on PopUp menu display

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When displaying a PopUp menu, sometimes the app crashes with this stack trace.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0xa8 pc=0x7ff7c26dcf7a]

goroutine 1 [running, locked to thread]:
fyne.io/fyne/v2/internal/widget.(*overlayRenderer).MinSize(0x7ff7c7571920?)
	fyne.io/fyne/v2@v2.6.2/internal/widget/overlay_container.go:96 +0x1a
fyne.io/fyne/v2/internal/widget.(*Base).MinSize(0xc000680460?)
	fyne.io/fyne/v2@v2.6.2/internal/widget/base.go:74 +0x2e
fyne.io/fyne/v2/widget.NewPopUpMenu(0xc0037382d0, {0x0, 0x0})
	fyne.io/fyne/v2@v2.6.2/widget/popup_menu.go:29 +0x1e5
fyne.io/fyne/v2/widget.ShowPopUpMenuAtPosition(0x7ff7c756db01?, {0x0?, 0x0?}, {0x6a6880?, 0xc0?})
	fyne.io/fyne/v2@v2.6.2/widget/popup_menu.go:40 +0x29
github.com/matwachich/fynex-widgets.(*Button).TappedSecondary(0xc0030e20f0, 0xc0038a1a30)
	github.com/matwachich/fynex-widgets@v0.0.0-20230706151007-895fe63e933d/button.go:116 +0xab
fyne.io/fyne/v2/internal/driver/glfw.(*window).processMouseClicked(0xc0000a3040, 0x2, 0x0, 0x0)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/window.go:527 +0x695
fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClicked(0xc0000a3040, 0xc001a51ab8?, 0x7ff7c2aba600?, 0x0, 0xc001a51a98?)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/window_desktop.go:410 +0x5d
github.com/go-gl/glfw/v3.3/glfw.goMouseButtonCB(0x7ff7c2206fac?, 0x1, 0x0, 0x0)
	github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20240506104042-037f3cc74f2a/input.go:333 +0x4e
github.com/go-gl/glfw/v3.3/glfw._Cfunc_glfwPollEvents()
	_cgo_gotypes.go:1544 +0x45
github.com/go-gl/glfw/v3.3/glfw.PollEvents()
	github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20240506104042-037f3cc74f2a/window.go:931 +0x13
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).pollEvents(...)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/loop_desktop.go:22
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0xc000e44000?)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/loop.go:152 +0x1aa
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0xc0002911e0)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/driver.go:162 +0x72
fyne.io/fyne/v2/app.(*fyneApp).Run(0xc000291290)
	fyne.io/fyne/v2@v2.6.2/app/app.go:77 +0x102
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0xc0000a3040)
	fyne.io/fyne/v2@v2.6.2/internal/driver/glfw/window.go:217 +0x64
main.main()
	github.com/matwachich/emr-made-simple/medecin/main.go:53 +0x108

```

### How to reproduce

Happens when displaying a PopUp menu, even on standard widget (DocTab tabs menu on the right).

### Screenshots

_No response_

### Example code

Could not reproduce the bug with simple code.

### Fyne version

2.6.1

### Go compiler version

1.21.1

### Operating system and version

Windows 11

### Additional Information

_No response_

--- a/container/tabs.go
+++ b/container/tabs.go
@@ -125,6 +125,11 @@ func tabsAdjustedLocation(l TabLocation, b baseTabs) TabLocation {
 func buildPopUpMenu(t baseTabs, button *widget.Button, items []*fyne.MenuItem) *widget.PopUpMenu {
 	d := fyne.CurrentApp().Driver()
 	c := d.CanvasForObject(button)
+	if c == nil {
+		// Overflow button detached from its canvas; nothing to host the
+		// pop-up on (see fyne-io/fyne#5965).
+		return nil
+	}
 	popUpMenu := widget.NewPopUpMenu(fyne.NewMenu("", items...), c)
 	buttonPos := d.AbsolutePositionForObject(button)
 	buttonSize := button.Size()

--- a/widget/date_entry.go
+++ b/widget/date_entry.go
@@ -149,6 +149,11 @@ func (e *DateEntry) setupDropDown() *Button {
 	}
 	dropDownButton := NewButton("", func() {
 		c := fyne.CurrentApp().Driver().CanvasForObject(e.super())
+		if c == nil {
+			// DateEntry detached from its canvas; cannot host calendar
+			// dropdown (see fyne-io/fyne#5965).
+			return
+		}
 
 		e.popUp = NewPopUp(e.dropDown, c)
 		e.popUp.ShowAtPosition(e.popUpPos())

--- a/widget/entry.go
+++ b/widget/entry.go
@@ -583,9 +583,15 @@ func (e *Entry) TappedSecondary(pe *fyne.PointEvent) {
 	}
 
 	driver := app.Driver()
+	c := driver.CanvasForObject(super)
+	if c == nil {
+		// Entry was detached from its canvas between the tap event and
+		// this call (see fyne-io/fyne#5965). Skip the context menu.
+		return
+	}
 	entryPos := driver.AbsolutePositionForObject(super)
 	popUpPos := entryPos.Add(pe.Position)
-	e.popUp = NewPopUpMenu(fyne.NewMenu("", menuItems...), driver.CanvasForObject(super))
+	e.popUp = NewPopUpMenu(fyne.NewMenu("", menuItems...), c)
 	e.popUp.ShowAtPosition(popUpPos)
 }
 

--- a/widget/entry_password.go
+++ b/widget/entry_password.go
@@ -50,7 +50,9 @@ func (r *passwordRevealer) Tapped(*fyne.PointEvent) {
 	r.entry.setFieldsAndRefresh(func() {
 		r.entry.Password = !r.entry.Password
 	})
-	fyne.CurrentApp().Driver().CanvasForObject(r).Focus(r.entry.super().(fyne.Focusable))
+	if c := fyne.CurrentApp().Driver().CanvasForObject(r); c != nil {
+		c.Focus(r.entry.super().(fyne.Focusable))
+	}
 }
 
 var _ fyne.WidgetRenderer = (*passwordRevealerRenderer)(nil)

--- a/widget/popup_menu.go
+++ b/widget/popup_menu.go
@@ -21,6 +21,9 @@ type PopUpMenu struct {
 //
 // Since: 2.0
 func NewPopUpMenu(menu *fyne.Menu, c fyne.Canvas) *PopUpMenu {
+	if c == nil {
+		return nil
+	}
 	m := &Menu{}
 	m.setMenu(menu)
 	p := &PopUpMenu{Menu: m, canvas: c}
@@ -40,6 +43,9 @@ func NewPopUpMenu(menu *fyne.Menu, c fyne.Canvas) *PopUpMenu {
 // It will automatically be positioned at the provided location and shown as an overlay on the specified canvas.
 func ShowPopUpMenuAtPosition(menu *fyne.Menu, c fyne.Canvas, pos fyne.Position) {
 	m := NewPopUpMenu(menu, c)
+	if m == nil {
+		return
+	}
 	m.ShowAtPosition(pos)
 }
 

--- a/widget/popup_test.go
+++ b/widget/popup_test.go
@@ -534,6 +534,59 @@ func TestModalPopUp_ResizeOnShow(t *testing.T) {
 	pop.Hide()
 }
 
+// Repro for fyne-io/fyne#5965.
+//
+// Root cause: internal/widget.overlayRenderer.MinSize dereferences
+// OverlayContainer.canvas unconditionally. When the canvas reference is nil
+// (e.g. the owning widget was detached from its window between the tap event
+// and the popup creation) the call panics with a nil pointer dereference.
+//
+// Higher-level trigger seen in the wild: widget.Select.Tapped ->
+// showPopUp -> NewPopUpMenu(menu, c) where Driver.CanvasForObject(s.super())
+// returns nil because the Select was already removed from its parent tree.
+// That path is driver-specific (real GLFW driver consults the canvas cache;
+// the test driver fakes CanvasForObject), so the underlying contract is
+// exercised here directly.
+func TestNewPopUpMenu_NilCanvas_NoPanic(t *testing.T) {
+	test.NewTempApp(t)
+
+	menu := fyne.NewMenu(
+		"",
+		fyne.NewMenuItem("a", func() {}),
+		fyne.NewMenuItem("b", func() {}),
+		fyne.NewMenuItem("c", func() {}),
+	)
+
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("NewPopUpMenu with nil canvas panicked: %v", r)
+		}
+	}()
+
+	pop := NewPopUpMenu(menu, nil)
+	if pop == nil {
+		return
+	}
+	pop.ShowAtPosition(fyne.NewPos(0, 0))
+}
+
+func TestShowPopUpMenuAtPosition_NilCanvas_NoPanic(t *testing.T) {
+	test.NewTempApp(t)
+
+	menu := fyne.NewMenu(
+		"",
+		fyne.NewMenuItem("only", func() {}),
+	)
+
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("ShowPopUpMenuAtPosition with nil canvas panicked: %v", r)
+		}
+	}()
+
+	ShowPopUpMenuAtPosition(menu, nil, fyne.NewPos(0, 0))
+}
+
 func TestModelPopUp_ResizeBeforeShow_CanvasSizeZero(t *testing.T) {
 	test.NewTempApp(t)
 

--- a/widget/select.go
+++ b/widget/select.go
@@ -280,6 +280,13 @@ func (s *Select) showPopUp() {
 	}
 
 	c := fyne.CurrentApp().Driver().CanvasForObject(s.super())
+	if c == nil {
+		// The Select was detached from its canvas (e.g. its parent
+		// container was rebuilt) between the tap event being delivered
+		// and this call. There is no canvas to host the pop-up, so do
+		// nothing rather than crash inside the overlay machinery.
+		return
+	}
 	pop := NewPopUpMenu(fyne.NewMenu("", items...), c)
 	pop.alignment = s.Alignment
 	pop.ShowAtPosition(s.popUpPos())

--- a/widget/select_entry.go
+++ b/widget/select_entry.go
@@ -97,6 +97,11 @@ func (e *SelectEntry) popUpPos() fyne.Position {
 func (e *SelectEntry) setupDropDown() *Button {
 	dropDownButton := NewButton("", func() {
 		c := fyne.CurrentApp().Driver().CanvasForObject(e.super())
+		if c == nil {
+			// SelectEntry detached from its canvas; nothing to host the
+			// dropdown on (see fyne-io/fyne#5965).
+			return
+		}
 
 		e.popUp = NewPopUpMenu(e.dropDown, c)
 		e.popUp.ShowAtPosition(e.popUpPos())

--- a/widget/select_test.go
+++ b/widget/select_test.go
@@ -3,11 +3,13 @@ package widget_test
 import (
 	"fmt"
 	"image"
+	"runtime/debug"
 	"testing"
 	"time"
 
 	"fyne.io/fyne/v2"
 	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
 	"fyne.io/fyne/v2/data/binding"
 	"fyne.io/fyne/v2/internal/cache"
 	"fyne.io/fyne/v2/layout"
@@ -610,6 +612,74 @@ func TestSelect_Layout(t *testing.T) {
 			assertRendersToPlatformMarkup(t, "select/%s/layout_"+name+".xml", window.Canvas())
 		})
 	}
+}
+
+// nilCanvasDriver wraps a fyne.Driver and returns nil from CanvasForObject
+// for one specific target, simulating the state observed on the real GLFW
+// driver after a widget is detached from its window between event delivery
+// and popup creation (fyne-io/fyne#5965). The stock test driver always
+// returns the last window's canvas regardless of the queried object, so the
+// detach scenario cannot otherwise be reproduced under the unit test driver.
+type nilCanvasDriver struct {
+	fyne.Driver
+	target fyne.CanvasObject
+}
+
+func (d *nilCanvasDriver) CanvasForObject(o fyne.CanvasObject) fyne.Canvas {
+	if o == d.target {
+		return nil
+	}
+	return d.Driver.CanvasForObject(o)
+}
+
+func (d *nilCanvasDriver) AbsolutePositionForObject(o fyne.CanvasObject) fyne.Position {
+	if o == d.target {
+		return fyne.NewPos(0, 0)
+	}
+	return d.Driver.AbsolutePositionForObject(o)
+}
+
+type wrappedApp struct {
+	fyne.App
+	drv fyne.Driver
+}
+
+func (a *wrappedApp) Driver() fyne.Driver { return a.drv }
+
+// Repro for fyne-io/fyne#5965: Select.Tapped must not panic when the widget
+// was already detached from its canvas (a frequent outcome of SetContent /
+// container rebuilds while a tap is in flight on the real driver).
+//
+// Failure mode on main: Select.Tapped -> showPopUp -> NewPopUpMenu(menu, nil)
+// -> OverlayContainer.Resize -> overlayRenderer.MinSize() -> nil deref on
+// OverlayContainer.canvas.
+func TestSelect_Tapped_AfterCanvasDetach_DoesNotPanic(t *testing.T) {
+	baseApp := test.NewTempApp(t)
+
+	sel := widget.NewSelect([]string{"a", "b", "c"}, nil)
+	w := test.NewWindow(container.NewVBox(sel))
+	defer w.Close()
+	w.Resize(fyne.NewSize(200, 200))
+
+	require.NotNil(t, fyne.CurrentApp().Driver().CanvasForObject(sel),
+		"precondition: Select is attached to a canvas before detach")
+
+	shim := &wrappedApp{App: baseApp, drv: &nilCanvasDriver{
+		Driver: baseApp.Driver(),
+		target: fyne.CanvasObject(sel),
+	}}
+	fyne.SetCurrentApp(shim)
+	t.Cleanup(func() { fyne.SetCurrentApp(baseApp) })
+
+	require.Nil(t, fyne.CurrentApp().Driver().CanvasForObject(fyne.CanvasObject(sel)),
+		"precondition: Select sees nil canvas after detach")
+
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Select.Tapped panicked after canvas detach: %v\n%s", r, debug.Stack())
+		}
+	}()
+	sel.Tapped(&fyne.PointEvent{Position: fyne.NewPos(0, 0)})
 }
 
 func assertRendersToPlatformMarkup(t *testing.T, file string, c fyne.Canvas) {

